src/app/search/search.component.ts

```diff
@@ -160,39 +160,51 @@ export class SearchComponent implements OnInit {
       this.filterEntry();
       this.toolSource.next(this.toolHits);
       this.workflowSource.next(this.workflowHits);
-      this.setupBuckets(hits);
+      this.setupAllBuckets(hits);
     });
   }
 
-  setupBuckets2(key, buckets: any) {
-    buckets.forEach(
-      bucket => {
-        if (this.buckets.get(key) == null) {
-          this.buckets.set(key, new Map<string, string>());
-          if (!this.setFilter) {
-            this.fullyExpandMap.set(key, false);
-            this.checkboxMap.set(key, new Map<string, boolean>());
-          }
-        }
-        this.buckets.get(key).set(bucket.key, bucket.doc_count);
+  /**
+   * Partially constructs the buckets, fullyExpandMap, and checkboxMap data structures
```

Comment:
Comments should give more insight into what we're trying to accomplish rather than mechanistically describing what is occurring. Why are these data structures being created?

Some good thoughts along these lines  https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/
https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/